# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build fontconfig-build glib-build libpng-build pixman-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

if PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists libxcb-build; then
  XCB=1
  REQUIRES="$REQUIRES libxcb-build"
fi

if PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists libxext-build libxrender-build; then
  XLIB=1
  REQUIRES="$REQUIRES libxext-build libxrender-build"
fi

test $XLIB || test $XCB || { echo Xlib or XCB packages required; exit 1; }

# configure (ac-2.68; am-1.11; lt-2.4.2)
libtoolize -c -f; aclocal -I build; autoconf; autoheader -f; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig LDFLAGS=-Wl,-rpath,/x11/lib ./configure --disable-interpreter --disable-static --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
install util/cairo-trace/cairo-trace $DESTDIR/x11/bin
install -d $DESTDIR/x11/include/cairo
install -m 644 cairo-version.h $DESTDIR/x11/include/cairo
install -m 644 src/cairo.h $DESTDIR/x11/include/cairo
install -m 644 src/cairo-deprecated.h $DESTDIR/x11/include/cairo
install -m 644 src/cairo-features.h $DESTDIR/x11/include/cairo
install -m 644 src/cairo-ft.h $DESTDIR/x11/include/cairo
install -m 644 src/cairo-pdf.h $DESTDIR/x11/include/cairo
install -m 644 src/cairo-ps.h $DESTDIR/x11/include/cairo
install -m 644 src/cairo-svg.h $DESTDIR/x11/include/cairo
test $XCB && install -m 644 src/cairo-xcb.h $DESTDIR/x11/include/cairo
test $XLIB && install -m 644 src/cairo-xlib.h $DESTDIR/x11/include/cairo
test $XLIB && install -m 644 src/cairo-xlib-xrender.h $DESTDIR/x11/include/cairo
install -m 644 util/cairo-gobject/cairo-gobject.h $DESTDIR/x11/include/cairo
install -d $DESTDIR/x11/lib
install src/.libs/libcairo.so.2.* $DESTDIR/x11/lib/libcairo.so.2
ln -sf libcairo.so.2 $DESTDIR/x11/lib/libcairo.so
install util/cairo-gobject/.libs/libcairo-gobject.so.2.* $DESTDIR/x11/lib/libcairo-gobject.so.2
ln -sf libcairo-gobject.so.2 $DESTDIR/x11/lib/libcairo-gobject.so
install -d $DESTDIR/x11/lib/cairo
install util/cairo-trace/.libs/libcairo-trace.so.0.0.0 $DESTDIR/x11/lib/cairo/libcairo-trace.so
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 src/cairo.pc $DESTDIR/x11/lib/pkgconfig
install -m 644 src/cairo-ft.pc $DESTDIR/x11/lib/pkgconfig
install -m 644 src/cairo-gobject.pc $DESTDIR/x11/lib/pkgconfig
install -m 644 src/cairo-png.pc $DESTDIR/x11/lib/pkgconfig
test $XCB && install -m 644 src/cairo-xcb.pc $DESTDIR/x11/lib/pkgconfig
test $XLIB && install -m 644 src/cairo-xlib.pc $DESTDIR/x11/lib/pkgconfig
test $XLIB && install -m 644 src/cairo-xlib-xrender.pc $DESTDIR/x11/lib/pkgconfig

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/cairo-build.pc << EOF
Name: cairo
Version: 1.12.18
Description: Cairo drawing library
Requires: $REQUIRES

devel=\\
/x11/include/cairo.h \\
/x11/include/cairo-deprecated.h \\
/x11/include/cairo-features.h \\
/x11/include/cairo-ft.h \\
/x11/include/cairo-gobject.h \\
/x11/include/cairo-pdf.h \\
/x11/include/cairo-ps.h \\
/x11/include/cairo-svg.h \\
/x11/include/cairo-svg.h \\
EOF
test $XCB && echo /x11/include/cairo-xcb.h \\ >> $DESTDIR/x11/share/pkgconfig/cairo-build.pc
test $XLIB && echo /x11/include/cairo-xlib.h \\ >> $DESTDIR/x11/share/pkgconfig/cairo-build.pc
test $XLIB && echo /x11/include/cairo-xlib-xrender.h \\ >> $DESTDIR/x11/share/pkgconfig/cairo-build.pc
cat >> $DESTDIR/x11/share/pkgconfig/cairo-build.pc << EOF
/x11/lib/libcairo.so \\
/x11/lib/libcairo-gobject.so \\
/x11/lib/pkgconfig/cairo.pc \\
/x11/lib/pkgconfig/cairo-ft.pc \\
/x11/lib/pkgconfig/cairo-gobject.pc \\
/x11/lib/pkgconfig/cairo-png.pc \\
EOF
test $XCB && echo /x11/lib/pkgconfig/cairo-xcb.pc \\ >> $DESTDIR/x11/share/pkgconfig/cairo-build.pc
test $XLIB && echo /x11/lib/pkgconfig/cairo-xlib.pc \\ >> $DESTDIR/x11/share/pkgconfig/cairo-build.pc
test $XLIB && echo /x11/lib/pkgconfig/cairo-xlib-xrender.pc >> $DESTDIR/x11/share/pkgconfig/cairo-build.pc
cat >> $DESTDIR/x11/share/pkgconfig/cairo-build.pc << EOF

exec=\\
/x11/bin/cairo-trace \\
/x11/lib/libcairo.so.2 \\
/x11/lib/libcairo-gobject.so.2 \\
/x11/lib/cairo/libcairo-trace.so
EOF

exit
--- cairo-1.12.18.orig/src/cairo-xcb-connection.c
+++ cairo-1.12.18/src/cairo-xcb-connection.c
@@ -38,7 +38,6 @@
 #include "cairo-list-inline.h"
 
 #include <xcb/xcbext.h>
-#include <xcb/bigreq.h>
 #include <errno.h>
 
 #if CAIRO_HAS_XCB_SHM_FUNCTIONS
@@ -663,7 +662,6 @@
     connection->flags = 0;
     connection->force_precision = -1;
 
-    xcb_prefetch_extension_data (xcb_connection, &xcb_big_requests_id);
     xcb_prefetch_extension_data (xcb_connection, &xcb_render_id);
 #if CAIRO_HAS_XCB_SHM_FUNCTIONS
     xcb_prefetch_extension_data (xcb_connection, &xcb_shm_id);
